src/server/graphql/models/Invitation/helpers.js

```diff
@@ -124,3 +126,27 @@ export const asyncInviteTeam = async (authToken, teamId, invitees) => {
   // throw errorObj({_error: 'Some invitations were not sent', type: 'inviteSendFail', failedEmails: inviteeErrors});
   // }
 };
+
+export const cancelInvitation = async (authToken, inviteId) => {
+  const r = getRethink();
+  const invite = r.table('Invitation').get(inviteId);
+  const {acceptedAt, teamId, tokenExpiration} = invite;
+  requireSUOrTeamMember(authToken, teamId);
+  const now = new Date();
+  const error = {};
+  if (acceptedAt) {
+    error.type = 'alreadyAccepted';
+  } else if (tokenExpiration < now) {
+    error.type = 'alreadyExpired';
+  }
+  if (error.type) {
+    // eslint-disable-next-line no-underscore-dangle
+    error._error = 'Cannot cancel invitation';
+    throw errorObj(error);
+  }
+  await r.table('Invitation').get(inviteId).update({
+    tokenExpiration: new Date(0),
```

Comment:
nonono, `Date` is a hot mess in JS. `new Date(0) !== Date(0)`. the first returns a date object of the epoch, the second returns a string of now.